src/persist-client/src/cfg.rs

```diff
@@ -329,10 +330,12 @@ pub struct DynamicConfig {
     batch_builder_max_outstanding_parts: AtomicUsize,
     blob_target_size: AtomicUsize,
     blob_cache_mem_limit_bytes: AtomicUsize,
+    pub(crate) compaction_enabled: AtomicBool,
```

Comment:
nit, but instead of pub(crate) let's add a set_compaction_enabled method so we don't leak the impl of DynamicConfig to the rest of the crate (though I think that's not necessary if we do my other comment)